### PR TITLE
Signupauth

### PR DESCRIPTION
**Pull request by hm2602**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,9 +11,12 @@
...
 /venv
+/app/.DS_Store
+
 
 # Secrets / Authentication
 secrets.py
 
 # SCSS
 .webassets-cache
+.DS_Store
```
Comment (alanhdu) [inline .gitignore]: Please remove these files from the git repo too.

File: app/forms.py [hunk 2]
```diff
@@ -9,9 +9,14 @@ class LoginForm(FlaskForm):
 class SignupForm(FlaskForm):
     firstname = StringField('First Name', [validators.Length(min=1, max=50)])
     lastname = StringField('Last Name', [validators.Length(min=1, max=50)])
-    email = StringField('Email Address', [validators.Email()])
+    email = StringField('Email Address', [
+        validators.Email(),
+        validators.Regexp('( |^)[^ ]*@(columbia|barnard|\w+\.(columbia))\.edu( |$)', message="Please fill in a Columbia-affiliated email"),
```
Comment (alanhdu) [inline app/forms.py]: I have no clue what's going on with the regex here.  We already have an email validator, so why don't we just check that the email ends w/ `barnard.edu` or `columbia.edu`:

```
.*@(columbia|barnard)\.edu$
```

Comment (Howon) [inline app/forms.py]: yeah this seems overly complicated

File: app/views.py [hunk 3]
```diff
@@ -53,12 +54,18 @@ def login():
         user = None
         try:
             user = auth.sign_in_with_email_and_password(form.email.data, form.password.data)
+            
+            if ():
```
Comment (jimmyro) [inline app/views.py]: what is this?  `if():`

File: app/views.py [hunk 4]
```diff
@@ -28,6 +28,7 @@ def signup():
 
         if user is not None:
             # verify email (form validator is just a regex!)
+
```
Comment (jimmyro) [inline app/views.py]: unnecessary newline

File: app/views.py [hunk 5]
```diff
@@ -53,12 +54,18 @@ def login():
...
+                flash('Please fill in a Columbia affiliated email address')
+
         except HTTPError: pass
 
         if user is not None:
             # if auth succeeds, see if email is verified
+
             accountInfo = auth.get_account_info(user['idToken'])
 
+
```
Comment (jimmyro) [inline app/views.py]: unnecessary newline

File: app/views.py [hunk 6]
```diff
@@ -53,12 +54,18 @@ def login():
         user = None
         try:
             user = auth.sign_in_with_email_and_password(form.email.data, form.password.data)
+            
+            if ():
+                flash('Please fill in a Columbia affiliated email address')
+
```
Comment (jimmyro) [inline app/views.py]: unnecessary newline

File: app/views.py [hunk 7]
```diff
@@ -53,12 +54,18 @@ def login():
...
         except HTTPError: pass
 
         if user is not None:
             # if auth succeeds, see if email is verified
+
```
Comment (jimmyro) [inline app/views.py]: unnecessary newline

File: app/forms.py [hunk 8]
```diff
@@ -9,9 +9,14 @@ class LoginForm(FlaskForm):
...
     firstname = StringField('First Name', [validators.Length(min=1, max=50)])
     lastname = StringField('Last Name', [validators.Length(min=1, max=50)])
-    email = StringField('Email Address', [validators.Email()])
+    email = StringField('Email Address', [
+        validators.Email(),
+        validators.Regexp('(columbia|barnard)\.edu', message="Please fill in a Columbia-affiliated email"),
+    ])
     password = PasswordField('New Password', [
         validators.DataRequired(),
+
```
Comment (jimmyro) [inline app/forms.py]: unnecessary newline

**Pull request closed without merging**